actionmailer/lib/action_mailer/test_helper.rb

```diff
@@ -151,7 +165,13 @@ def assert_enqueued_email_with(mailer, method, args: nil, queue: "mailers", &blo
     #     end
     #   end
     def assert_no_enqueued_emails(&block)
-      assert_no_enqueued_jobs only: [ ActionMailer::DeliveryJob, ActionMailer::Parameterized::DeliveryJob ], &block
+      assert_enqueued_emails(0, &block)
     end
+
+    private
+
+      def queue_adapter_for_test
+        ActionMailer::TestAdapter.new
```

Comment:
The mailer_adapter has the precedence, active_job assertions would just behave as usual though.

Reply:
Right, I can see regular job assertions run through `ActionMailer::TestAdapter` now. I'm not sure I'm a huge fan of that. Do you think it makes sense to change this behaviour?

Reply:
Having 2 adapters is even more confusing IMO. If we want to avoid having an extra adapter to fix this issue I think we don't have the choice but to implement lambda filter. @rafaelfranca got some opinion/advice on this one ?

Reply:
Allowing the filter to be a lambda and building the lambda in `assert_enqueued_emails` and friends makes more sense to me